qemu/tests/qemu_disk_img.py

```diff
@@ -99,6 +99,15 @@ def __md5sum(self, dst):
     def save_file(self, dst):
         error.context("save file('%s') md5sum in guest" % dst, logging.info)
         self.__create_file(dst)
+        login_timeout = int(self.params.get("login_timeout", 360))
+        session = self.vm.wait_for_login(timeout=login_timeout)
+        error.context("sync guest data")
+        cmd = "sync"
+        status, output = session.cmd_status_output(cmd)
+        if status != 0:
+            logging.error("Execute '%s' with failures('%s') " % (cmd, output))
+            return None
```

Comment:
When guest shutdown abnormally like quit by qmp after copy a file from host to guest, the file may not be flushed to disk which will make the test failed. So we need to do sync here . If the command failed to issue, we need to raise TestFail exception.
One simple way is use session.cmd('sync'). if the command failed, exception ShellCmdError  will be raised.
The other way is as below:
status, output = session.cmd_status_output(cmd)
if status != 0:
    raise error.TestFail("Execute '%s' with failures('%s') " % (cmd, output))